pipeline/k8s_autoscale.go

```diff
@@ -0,0 +1,167 @@
+package pipeline
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/influxdata/kapacitor/services/k8s/client"
+	"github.com/influxdata/kapacitor/tick/ast"
+)
+
+const (
+	DefaultNamespaceTag = "namespace"
+	DefaultKindTag      = "kind"
+	DefaultResourceTag  = "resource"
+)
+
+// K8sAutoscaleNode triggers autoscale events for a resource on a Kubernetes cluster.
+// The node also outputs points for the triggered events.
+//
+// Example:
+//     // Target 100 requests per second per host
+//     var target = 100.0
+//     var min = 1
+//     var max = 100
+//     var period = 5m
+//     var every = period
+//     stream
+//         |from()
+//             .measurement('requests')
+//             .groupBy('host', 'deployment')
+//             .truncate(1s)
+//         |derivative('value')
+//             .as('requests_per_second')
+//             .unit(1s)
+//             .nonNegative()
+//         |groupBy('deployment')
+//         |sum('requests_per_second')
+//             .as('total_requests')
+//         |window()
+//             .period(period)
+//             .every(every)
+//         |mean('total_requests')
+//             .as('total_requests')
+//         |k8sAutoscale()
+//             // Get the name of the deployment from the 'deployment' tag.
+//             .resourceNameTag('deployment')
+//             .min(min)
+//             .max(max)
+//             // Set the desired number of replicas based on target.
+//             .replicas(lambda: int(ceil("total_requests" / target)))
+//         |influxDBOut()
+//             .database('deployments')
+//             .measurement('scale_events')
+//             .precision('s')
```

Comment:
@goller Does this TICkscript make sense to you, specifically the `|k8sAutoscale()` part?

Want to make sure this API fits what you would expect using k8s.